prov/gni/test/cntr.c

```diff
@@ -797,3 +797,89 @@ Test(cntr_mt, write_wait)
 
 	dbg_printf("done\n");
 }
+
+void *do_add_cntr(void *arg)
+{
+	int i = 0, ret, iters = ((int *)arg)[0];
+
+	for (; i < iters; i++) {
+		ret = fi_cntr_add(ep_write_cntrs[0], 1);
+		cr_assert(ret == FI_SUCCESS);
```

Comment:
Ok, I'll need a lock around `fi_cntr_add` and `fi_cntr_read` for these mt tests then. But, that will not test the thread safety of the `fi_cntr_add` interface. Perhaps we should have a single threaded test for this (something added to the `cntr` test suite)?